### PR TITLE
[Feature]  load map column from parquet

### PR DESCRIPTION
**Pull request by fzhedu**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：
<!--
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

## Problem Summary(Required) ：
<!-- (Please describe the changes you have made. In which scenarios will this bug be triggered and what measures have you taken to fix the bug?) -->
follow https://github.com/StarRocks/starrocks/pull/21411, load map column from parquet, and remove duplicated keys when loading map columns.

```
mysql> LOAD LABEL tmapp(DATA INFILE("hdfs://172.26.194.238:9000/starrocks/test_data/map/file_reader_test_map.parquet") INTO TABLE tmapp FORMAT AS "parquet" (c1,c2,c3,c4))WITH BROKER;
Query OK, 0 rows affected (0.01 sec)

mysql> SHOW LOAD where label =  "tmapp";
+-------+-------+-----------+-------------------+--------+----------+---------+------------------------------------------------------+-------------------------------------------------------------------------------------------------------------+---------------------+---------------------+---------------------+---------------------+---------------------+-------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| JobId | Label | State     | Progress          | Type   | Priority | EtlInfo | TaskInfo                                             | ErrorMsg                                                                                                    | CreateTime          | EtlStartTime        | EtlFinishTime       | LoadStartTime       | LoadFinishTime      | TrackingSQL | JobDetails                                                                                                                                                                                                                                                        |
+-------+-------+-----------+-------------------+--------+----------+---------+------------------------------------------------------+-------------------------------------------------------------------------------------------------------------+---------------------+---------------------+---------------------+---------------------+---------------------+-------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| 53187 | tmapp | FINISHED  | ETL:100%; LOAD:100% | BROKER | NORMAL   | unselected.rows=0; dpp.abnorm.ALL=0; dpp.norm.ALL=8 | resource:N/A; timeout(s):14400; max_filter_ratio:0.0 | NULL                                                                                                        | 2023-04-12 14:59:12 | 2023-04-12 14:59:15 | 2023-04-12 14:59:15 | 2023-04-12 14:59:15 | 2023-04-12 14:59:19 |             | {"All backends":{"9beac059-93ee-4a00-b0cb-d55df13d101a":[37027]},"FileNumber":1,"FileSize":1727,"InternalTableLoadBytes":1011,"InternalTableLoadRows":8,"ScanBytes":1727,"ScanRows":8,"TaskNumber":1,"Unfinished backends":{"9beac059-93ee-4a00-b0cb-d55df13d101a":[]}} |
+-------+-------+-----------+---------------------+--------+----------+-----------------------------------------------------+------------------------------------------------------+-------------------------------------------------------------------------------------------------------------+---------------------+---------------------+---------------------+---------------------+---------------------+-------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
3 rows in set (0.00 sec)

mysql> select * from tmapp;
+------+------------------------+---------------------------------------------+---------------------+
| c1   | c2                     | c3                                          | c4                  |
+------+------------------------+---------------------------------------------+---------------------+
|    3 | {"k2":2,"k3":3,"k5":4} | {"e1":{"f1":1,"f2":2,"f3":3}}               | {"g1":[1,2,3]}      |
|    5 | {"k3":4}               | {"e2":{"f2":2}}                             | {"g1":[null]}       |
|    6 | {"k1":null}            | {"e2":{"f2":null}}                          | {"g1":[1]}          |
|    1 | {"k1":0,"k2":1}        | {"e1":{"f1":1,"f2":2}}                      | {"g1":[1,2]}        |
|    2 | {"k1":1,"k3":2,"k4":3} | {"e1":{"f1":1,"f2":2},"e2":{"f1":1,"f2":2}} | {"g1":[1],"g3":[2]} |
|    4 | {"k1":3,"k2":4,"k3":5} | {"e2":{"f2":2}}                             | {"g1":[1]}          |
|    7 | {"k1":1,"k2":2}        | {"e2":{"f2":2}}                             | {"g1":[1,2,3]}      |
|    8 | {"k3":4}               | {"e1":{"f1":1,"f2":2,"f3":3}}               | {"g2":[1],"g3":[2]} |
+------+------------------------+---------------------------------------------+---------------------+
8 rows in set (0.01 sec)

mysql> desc tmapp;
+-------+---------------------------------------------+------+-------+---------+-------+
| Field | Type                                        | Null | Key   | Default | Extra |
+-------+---------------------------------------------+------+-------+---------+-------+
| c1    | int                                         | YES  | true  | NULL    |       |
| c2    | map<varchar(65533),int>                     | YES  | false | NULL    |       |
| c3    | map<varchar(65533),map<varchar(65533),int>> | YES  | false | NULL    |       |
| c4    | map<varchar(65533),array<int>>              | YES  | false | NULL    |       |
+-------+---------------------------------------------+------+-------+---------+-------+
4 rows in set (0.00 sec)
```

### performance
load 44GB data from hive to SR.

```
2023-04-17 11:14:10 | 2023-04-17 11:36:18 to_json costs 22 min 8sec
2023-04-14 17:47:32 | 2023-04-14 18:01:39 to_map costs 14min 7 sec
```

### TODO
make sure the map->keys should be unique.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: be/src/formats/parquet/column_reader.cpp [hunk 1]
```diff
@@ -275,6 +275,7 @@ class MapColumnReader : public ColumnReader {
         if (_value_reader == nullptr) {
             value_column->append_default(offsets.back());
         }
+        map_column->remove_duplicated_keys();
```
Comment (zombee0) [inline be/src/formats/parquet/column_reader.cpp:279]: How about later materialized that all key is null?

**Review by murphyatwork — DISMISSED** [review]

File: be/test/formats/parquet/file_reader_test.cpp [hunk 2]
```diff
@@ -1669,16 +1669,14 @@ TEST_F(FileReaderTest, TestReadMapColumnWithPartialMaterialize) {
...
+    EXPECT_EQ(chunk->debug_row(1), "[2, {'k1':NULL,'k3':NULL,'k4':NULL}, {NULL:{'f1':NULL,'f2':NULL}}, {NULL:[2]}]");
     EXPECT_EQ(chunk->debug_row(2),
               "[3, {'k2':NULL,'k3':NULL,'k5':NULL}, {NULL:{'f1':NULL,'f2':NULL,'f3':NULL}}, {NULL:[1,2,3]}]");
     EXPECT_EQ(chunk->debug_row(3), "[4, {'k1':NULL,'k2':NULL,'k3':NULL}, {NULL:{'f2':NULL}}, {NULL:[1]}]");
     EXPECT_EQ(chunk->debug_row(4), "[5, {'k3':NULL}, {NULL:{'f2':NULL}}, {NULL:[NULL]}]");
     EXPECT_EQ(chunk->debug_row(5), "[6, {'k1':NULL}, {NULL:{'f2':NULL}}, {NULL:[1]}]");
     EXPECT_EQ(chunk->debug_row(6), "[7, {'k1':NULL,'k2':NULL}, {NULL:{'f2':NULL}}, {NULL:[1,2,3]}]");
-    EXPECT_EQ(chunk->debug_row(7), "[8, {'k3':NULL}, {NULL:{'f1':NULL,'f2':NULL,'f3':NULL}}, {NULL:[1],NULL:[2]}]");
+    EXPECT_EQ(chunk->debug_row(7), "[8, {'k3':NULL}, {NULL:{'f1':NULL,'f2':NULL,'f3':NULL}}, {NULL:[2]}]");
 }
```
[the remark this answers is not in the document]
Comment (Smith-Cruise) [reply be/test/formats/parquet/file_reader_test.cpp:1680]: Could you add this comment in code?

File: be/test/formats/parquet/file_reader_test.cpp [hunk 3]
```diff
@@ -1669,16 +1669,14 @@ TEST_F(FileReaderTest, TestReadMapColumnWithPartialMaterialize) {
     ASSERT_TRUE(status.ok());
     EXPECT_EQ(chunk->num_rows(), 8);
     EXPECT_EQ(chunk->debug_row(0), "[1, {'k1':NULL,'k2':NULL}, {NULL:{'f1':NULL,'f2':NULL}}, {NULL:[1,2]}]");
-    EXPECT_EQ(chunk->debug_row(1),
-              "[2, {'k1':NULL,'k3':NULL,'k4':NULL}, {NULL:{'f1':NULL,'f2':NULL},NULL:{'f1':NULL,'f2':NULL}}, "
-              "{NULL:[1],NULL:[2]}]");
+    EXPECT_EQ(chunk->debug_row(1), "[2, {'k1':NULL,'k3':NULL,'k4':NULL}, {NULL:{'f1':NULL,'f2':NULL}}, {NULL:[2]}]");
```
Comment (Smith-Cruise) [inline be/test/formats/parquet/file_reader_test.cpp:1672]: add comment

**Review by Smith-Cruise — COMMENTED** [review]
Remains LGTM

File: be/src/formats/parquet/column_reader.cpp (hunk 1 above)
Comment (fzhedu) [reply be/src/formats/parquet/column_reader.cpp]: let it be a TODO

**Review by murphyatwork — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Pull request merged**